components/prism-tt2.js

```diff
@@ -0,0 +1,78 @@
+(function(Prism) {
+
+	var tt2_pattern = /\[%[\s\S]+?%\]|^%%*?/;
+
+	Prism.languages.tt2 = Prism.languages.extend('markup', {
+		'tt2': {
+			pattern: tt2_pattern,
+			inside: {
+				'comment': {
+					pattern: /(^|[^\\$])#.*/,
+					lookbehind: true
+				},
+				'delimiter': {
+					pattern: /^(?:\[%|%%)-?|-?%]$/i,
+					alias: 'punctuation'
+				},
+				'string': /(["'])(?:\\.|(?!\1)[^\\\r\n])*\1/,
+				'number': /\b0x[\dA-Fa-f]+\b|(?:\b\d+\.?\d*|\B\.\d+)(?:[Ee][+-]?\d+)?/,
+				'keyword': /\b(?:GET|CALL|SET|DEFAULT|INSERT|INCLUDE|PROCESS|WRAPPER|BLOCK|IF|UNLESS|ELSIF|ELSE|SWITCH|CASE|FOREACH|WHILE|FILTER|USE|MACRO|RAWPERL|PERL|TRY|THROW|CATCH|FINAL|NEXT|LAST|RETURN|STOP|CLEAR|META|TAGS|DEBUG)\b/,
+				'operator': /\b(?:=>|==|!=|<|<=|>|>=|&&\|\|!|and|or|not)\b/,
+				'punctuation': /[[\]{},]/,
+				'variable': /[a-zA-Z][\.\/\$a-zA-Z0-9]*/
```

Comment:
You don't need those three escapes in the caracter class. Also, you could benefit from the `i` flag here to remove the `A-Z` occurrences, and could use `\d` instead of `0-9`.

Reply:
That's my personal style but I can follow your convention.  Off-topic, but this is my rationale:

- The extraneous escapes help both syntax highlighters and inexperienced readers.
- `/[a-zA-Z]/` is slightly more efficient than `/[a-z]/i`.  Since the `i` modifier is unicode aware, case-insensitive matching is potentially expensive.
- Does `\d` match "۴۲" or not? It does not.  But why make people figure, when you can be explicit.

Yes, rather academic, and I have no problem following your style instead.

Reply:
Haha, you're right that I usually go for the shortest form instead, at the cost of a tiny bit of readability. But hey, regexps _are_ quite hard to read anyway. ^^

(Note that regarding your unicode-related points, I'm pretty sure they do not apply unless the `u` flag is used, if they even do.)

Reply:
True.  The `u` makes a difference in this case:

```javascript
console.log("\u212a".toLowerCase());
console.log("\u212a".match(/[a-z]/iu));
```

Although the behavior without `/u` could also be qualified as a bug.

But, honestly, do you know whether "۲" (an Arabic two) matches `/d` or whether "\u3000" (ideographic space) matches `\s` (without `/u`)?  The Arabic digit does not match `\d` but ideographic space matches `\s`.

I think the usage in Prism is correct everywhere so this is rather academic and it was really not meant as criticism.

Reply:
No worries.